DemoApp/DemoAppCoordinator.swift

```diff
@@ -99,6 +99,8 @@ final class DemoAppCoordinator: NSObject, UNUserNotificationCenterDelegate {
         Components.default.channelListRouter = DemoChatChannelListRouter.self
         Components.default.channelVC = CustomChannelVC.self
         Components.default.messageContentView = CustomMessageContentView.self
+        Components.default.messageListVC.isDateSeparatorEnabled = true
+        Components.default.messageListVC.isDateOverlayEnabled = true
```

Comment:
IMHO these shouldn't be in `Components`, it's not like the pattern we've been using. A property on the actual `MessageListVC` would suffice?

Reply:
We also have a `Components.default.mentionAllAppUsers` which is also a flag. With this approach, at least the flag is set only in the context of the message list. And ofc putting multiple flags directly on the component's config will not scale very well. I did it in this way because we have been having feedback that requiring to subclassing just to turn on/off a flag is not very dev friendly.

Reply:
One limitation of the current approach is that the config will be always the same for Channels and Threads, you can't provide a different config for each of them since the properties are static... So for that reason, it might make sense to switch to instance property like we are doing in other components, even tho its hard to change :/ We can also add a property directly to `Components` and also overridable by instance. With this, we get the best of both worlds. The idea here is to make it easier to provide these turn on/off these configs without subclassing, since multiple customers have provided feedback about this

Reply:
@b-onc I've updated the approach to be more flexible. Both approaches are now supported.